src/wp-includes/post.php

```diff
@@ -8018,3 +8019,37 @@ function use_block_editor_for_post_type( $post_type ) {
 	 */
 	return apply_filters( 'use_block_editor_for_post_type', true, $post_type );
 }
+
+/**
+ * Registers any additional post meta fields.
+ *
+ * @since 6.3.0 Adds sync_status meta field to the wp_block post type so an unsynced option can be added.
+ *
+ * @link https://github.com/WordPress/gutenberg/pull/51144
+ *
+ */
+function wp_register_initial_post_meta() {
+	$post_type = 'wp_block';
+	register_post_meta(
+		$post_type,
+		'sync_status',
+		array(
+			'auth_callback'     => function() {
+				return current_user_can( 'edit_posts' );
+			},
```

Comment:
This is unnecessary since we already check that the user has permission to edit the relevant post. An `auth_callback` let's us apply further restrictions.

Reply:
thanks, have removed.

Reply:
@TimothyBJacobs a follow up question to this [from here](https://github.com/WordPress/gutenberg/pull/51144#discussion_r1242922720), about whether we should be should be doing any postmeta capability check?